### PR TITLE
Fix empty filename case

### PR DESCRIPTION
**Pull request by PaDarochek**

### CODE REVIEW

File: casr/src/bin/casr-cli.rs [hunk 1]
```diff
@@ -824,12 +824,16 @@ fn print_summary(dir: &Path, unique_crash_line: bool) {
             .map(|res| res.unwrap().path())
             .any(|e| e.extension().is_some() && e.extension().unwrap() == "casrep")
     {
-        clusters.push((dir.to_path_buf(), 0));
```
Comment (SweetVishnya) [inline casr/src/bin/casr-cli.rs:827]: canonicalize may fail

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: casr/src/bin/casr-cli.rs (hunk 1 above)
Comment (PaDarochek) [reply casr/src/bin/casr-cli.rs:827]: Fixed, just try to canonicalize, push `dir` itself if failed

**Review by Avgor46 — APPROVED** [review]

File: casr/src/bin/casr-cli.rs [hunk 2]
```diff
@@ -824,12 +824,20 @@ fn print_summary(dir: &Path, unique_crash_line: bool) {
...
+        }
     }
 
     for (clpath, _) in clusters {
         let cluster = clpath.as_path();
-        let filename = cluster.file_name().unwrap().to_str().unwrap();
+        let filename = if let Some(cl_filename) = cluster.file_name() {
+            cl_filename.to_str().unwrap()
+        } else {
+            cluster.to_str().unwrap()
```
Comment (SweetVishnya) [inline casr/src/bin/casr-cli.rs:839]: why should be use full path here?

Comment (PaDarochek) [reply casr/src/bin/casr-cli.rs:839]: If the filename is empty it means that it ends with '.' or is '/' and we couldn't canonicalize it. Then we have no other options besides using this path as is.

**Review by hkctkuy — APPROVED** [review]

File: casr/src/bin/casr-cli.rs (hunk 2 above)
Comment (anfedotoff) [reply casr/src/bin/casr-cli.rs:839]: > If the filename is empty it means that it ends with '.' or is '/' and we couldn't canonicalize it. Then we have no other options besides using this path as is.

Could you provide some use cases, when and why this situation happens?
It seems to me, that [this code](https://github.com/ispras/casr/blob/master/casr/src/bin/casr-cli.rs#L813) filter directories.

Comment (PaDarochek) [reply casr/src/bin/casr-cli.rs:839]: The case is when casr-cli is called with an argument of directory name, and that directory contains raw casreps. For example, if casreps are located in root directory (a poked-out case, but still), or in cwd '.' and we couldn't canonicalize it for some reason. The bug was detected when calling casr-cli on cwd with raw reports.

Comment (SweetVishnya) [reply casr/src/bin/casr-cli.rs:839]: Can we add a comment in code stating this?

**Review by SweetVishnya — APPROVED** [review]
Just needs some comments and ready to merge

File: casr/src/bin/casr-cli.rs (hunk 2 above)
Comment (PaDarochek) [reply casr/src/bin/casr-cli.rs:842]: Done

**Pull request merged**